<?xml version="1.0"?>

<launch>

  <!-- start the drive by write (DBW) system -->
  <arg name="sys" default="true"/>
  <group if="$(arg sys)">
    <include file="$(find dbw_polaris_can)/launch/dbw.launch" />
  </group>

  <!-- connect to the Polaris GEM e2's camera -->
  <include file="$(find avt_vimba_camera)/launch/mono_camera.launch" />

  <!-- launch the dynamic reconfigure gui -->
  <node pkg="rqt_reconfigure" name="rqt_console" type="rqt_reconfigure" args="-t"/>

  <!-- run the yellow blob detection script -->
  <node pkg="team_triangle" name="detect_yellow" type="detect_yellow.py" required="true" output="screen">
    <param name="imgtopic_name" type="str" value="camera_view" />
  </node>

  <!-- run the line following algorithm script -->
  <node pkg="team_triangle" name="follow_line" type="follow_line.py" required="true" output="screen">
    <param name="imgtopic_name" type="str" value="camera_view" />
  </node>

  <!-- run the follow line evaluation script -->
  <node pkg="team_triangle" name="follow_line_eval" type="follow_line_eval.py" required="true" output="screen">
    <param name="imgtopic_name" type="str" value="camera_view" />
  </node>

</launch>
